File: favourite-lyrics-core/src/main/java/org/jesperancinha/lyrics/core/service/LyricsService.java
package org.jesperancinha.lyrics.core.service;

import org.jesperancinha.lyrics.domain.data.LyricsDto;
import org.jesperancinha.lyrics.domain.data.LyricsFullDto;

import java.util.List;
import java.util.UUID;

public interface LyricsService {

    void addLyrics(LyricsDto lyricsDto);

    void removeLyrics(LyricsDto lyricsDto);

    void updateLyrics(LyricsDto lyricsDto);

    List<LyricsDto> getAllLyrics();

    List<LyricsFullDto> getAllFullLyrics();

    LyricsDto getLyricsById(UUID lyricsId);

}

File: favourite-lyrics-core/src/main/kotlin/org/jesperancinha/lyrics/core/service/LyricsService.kt
package org.jesperancinha.lyrics.core.service

import org.jesperancinha.lyrics.domain.data.LyricsDto
import org.jesperancinha.lyrics.domain.data.LyricsFullDto
import java.util.*

interface LyricsService {
    fun addLyrics(lyricsDto: LyricsDto)
    fun removeLyrics(lyricsDto: LyricsDto)
    fun updateLyrics(lyricsDto: LyricsDto)
    fun getAllLyrics(): List<LyricsDto>
    fun getAllFullLyrics(): List<LyricsFullDto>
    fun getLyricsById(lyricsId: UUID): LyricsDto
}